lib/addons/core-addon/index.ts

```diff
@@ -14,15 +14,19 @@ export class CoreAddOnProps {
      * Version of the add-on to use. Must match the version of the cluster where it
      * will be deployed it
      */
-    readonly version: string;
+    readonly version?: string;
```

Comment:
why making it optional? I thought version is mandatory. 
Does not lack of version make deployment non-deterministic?
So far we had it required, otherwise you deployment becomes dependent on the point in time. We can discuss more.

Reply:
Except for the `kube-proxy` and `vpc-cni` addon, rest of the Addons such as `ebs`, `coredns`, `adot` EKS addons are already updated to remove `version` field. Keeping `version` field optional lets the addon give the flexbility to pick the latest version during the setup. IMO we should keep it optional and once EKS Addon's stabilizes this field should be completely removed.

Reply:
the issue with the latest version is that customer has no idea what version was picked up for a particular cluster. 
Agree to keep it optional in case customers do need this behavior. I would not recommend it for production though.